libc/src/stdio/baremetal/fgets.cpp

```diff
@@ -0,0 +1,49 @@
+//===-- Implementation of fgets for baremetal -------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/stdio/fgets.h"
+
+#include "hdr/stdio_macros.h" // for EOF.
+#include "hdr/types/FILE.h"
+#include "src/__support/common.h"
+#include "src/__support/libc_errno.h"
+#include "src/__support/macros/config.h"
+#include "src/stdio/baremetal/file_internal.h"
+
+namespace LIBC_NAMESPACE_DECL {
+
+LLVM_LIBC_FUNCTION(char *, fgets,
+                   (char *__restrict str, int count,
+                    ::FILE *__restrict stream)) {
+  if (count < 1)
+    return nullptr;
+
+  unsigned char c = '\0';
```

Comment:
Why is this unsigned? Using a plain char would avoid the need for reinterpret_cast and also avoid the implicit conversion when assigning to str.

Reply:
I just followed the generic implementation but I don't know the reason why we used `unsigned char` there and I agree that using just `char` is simpler.